### PR TITLE
Parallel job submitting test

### PR DESCRIPTION
**Pull request by Mykhailenko**
TRELLO https://trello.com/c/SNBYK90S/2983-benchmark-azure-make-api-responsiveness-1s

Basically, to improve API Responsiveness, we have decided to add ImmediateService annotation to  submit(job) method of SchedulerFrontend. 

Before I will add this annotation I would like to merge this PR, where I add:
- functional test TestJobSubmittedParallel 
- performance test AverageJobSubmittingTimeTest

### CODE REVIEW

File: scheduler/scheduler-server/src/test/java/functionaltests/job/TestJobSubmittedParallel.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        // wait until all jobs are finished
+        silentMap(futures.stream(), future -> future.get()).forEach(col -> allJobId.addAll(col));
+
+        int EXPECTER_NUMBER_OF_JOBS = THREAD_POOL_SIZE * NUMBER_OF_JOBS_PER_THREAD;
+
+        assertEquals(EXPECTER_NUMBER_OF_JOBS, allJobId.size());
+
+        assertEquals(EXPECTER_NUMBER_OF_JOBS, GetTotalJobCount());
+
```
Comment (fviale) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/TestJobSubmittedParallel.java]: you need to test as well that you can safely interact with the job right after submission, for example by calling schedulerHelper.getSchedulerInterface().getJobState() and some method inside (e.g. getStartTime())

Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/functionaltests/job/TestJobSubmittedParallel.java]: Roger that

File: scheduler/scheduler-server/src/test/java/performancetests/metrics/ParallelSequentionalJobSubmition.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        final ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+
+        long startTime = System.currentTimeMillis();
+        List<Future<List<JobId>>> futures = new ArrayList<>();
+        for (int i = 0; i < threadPoolSize; ++i) {
+            final Future<List<JobId>> future = executorService.submit(new Callable<List<JobId>>() {
+                @Override
+                public List<JobId> call() throws Exception {
+                    List<JobId> result = new ArrayList<>();
```
Comment (MaelAudren) [inline scheduler/scheduler-server/src/test/java/performancetests/metrics/ParallelSequentionalJobSubmition.java]: List<JobId> result = new ArrayList<>(); 
=> List<JobId> result = new ArrayList<>(numberOfJobSubmittedByThread);

Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/performancetests/metrics/ParallelSequentionalJobSubmition.java]: Thanks

**Comment by Mykhailenko** [comment]
Retest this please

**Comment by Mykhailenko** [comment]
Retest this please

**Pull request merged**